dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/ProgramStageWorkingListObjectBundleHookTest.java

```diff
@@ -149,10 +156,8 @@ void shouldFailWhenDatePeriodAbsoluteAndNoStartNorEndDateSpecifiedForEnrollmentO
 
         List<ErrorReport> errorReports = workingListHook.validate( programStageWorkingList, bundle );
 
-        assertEquals( 1, errorReports.size() );
-        assertEquals(
-            "Start date or end date have to be specified when date period type is set to ABSOLUTE for item `EnrollmentIncidentDate`",
-            errorReports.get( 0 ).getMessage() );
+        ErrorReport expectedErrorReport = new ErrorReport( this.getClass(), E4062, "EnrollmentIncidentDate" );
```

Comment:
Does the value of the first parameter `this.getClass()` matter to our test? If not we could set it to null. If that would fail because its accessed I would wrap that in a helper to highlight whats important like `error(E4062, "EnrollmentIncidentDate")`. Hiding the fact that we have to pass `this.getClass()`.

The signatures you chose in `assertErrorReport` taking an expected error hides a bit what you are asserting on and it forces us to repeat a bit what we do in our implementation.

For these reasons I rather use something like

`assertHasError(errors, Code code, String messageContains )`


So I don't have to deal with creating an error in the test and figuring out which fields I need to set on the error 😅 

Maybe others have a different view on this 😄 👌🏻

Reply:
Makes sense to have the test as much decoupled as possible from the implementation.